### ISSUE TITLE
Analyzer warnings when using the remainder function

### ISSUE THREAD

**Issue by a-siva** [dart-lang/sdk#40352]
When migrating the below code to NNBD using pkg/dev_compiler/tool/check_nnbd_sdk.dart tool, the analyzer produces a warning (requires an explicit cast)

```
const int DAYS_IN_4_YEARS = 4 * 365 + 1;	
const int DAYS_IN_100_YEARS = 25 * DAYS_IN_4_YEARS - 1;
const int DAYS_IN_100_YEARS = 25 * DAYS_IN_4_YEARS - 1;
const int DAYS_IN_400_YEARS = 4 * DAYS_IN_100_YEARS + 1;
const int daysSince1970 = 200;
const in DAYS_IN_400_YEARS
int days = daysSince1970;
days = days.remainder(DAYS_IN_400_YEARS);

ERROR|STATIC_TYPE_WARNING|INVALID_ASSIGNMENT|lib/core/core.dart|2092|12|33|A value of type 'num' can't be assigned to a variable of type 'int'.
```

It requires the user to do an explicit cast which can be inefficient.

We need a mechanism that would not require the user to make an explicit cast here.

**Comment by stereotype441** [comment]
I believe the analyzer is correctly implementing the language specification here, since the declared return type of `remainder` is `num`, and implicit downcasts are not allowed with NNBD.  So I'm reclassifying as a language issue.  @leafpetersen FYI

**Comment by a-siva** [comment]
Yes the analyzer is following the spec here, the issue of no implicit downcasts with NNBD makes code that was clean before to start issuing warnings and people might go and fix their code by adding explicit casts that could potentially have performance implications.

**Comment by srawlins** [comment]
I'll move this to the language repo since the "area-language" tag is marked deprecated; new issues should go to that repo.
